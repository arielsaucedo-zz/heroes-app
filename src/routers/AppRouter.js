import React from "react";
import { BrowserRouter as Router, Switch } from "react-router-dom";

import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";
import { LoginScreen } from "../components/auth/LoginScreen";
import { HeroesScreen } from "../components/heroes/HeroesScreen";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute exact path="/login" component={LoginScreen} />
          <PrivateRoute exact path="/" component={HeroesScreen} />
        </Switch>
      </div>
    </Router>
  );
};
