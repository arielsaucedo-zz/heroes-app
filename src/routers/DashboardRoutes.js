import React from "react";
import { Switch, Route, Redirect } from "react-router";

import { HeroesScreen } from "../components/heroes/HeroesScreen";
import { SearchScreen } from "../components/search/SearchScreen";
import { Navbar } from "../components/ui/Navbar";

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-2">
        <Switch>
          <Route exact path="/heroes" component={HeroesScreen} />
          <Route exact path="/search" component={SearchScreen} />

          <Redirect to="/heroes" />
        </Switch>
      </div>
    </>
  );
};
