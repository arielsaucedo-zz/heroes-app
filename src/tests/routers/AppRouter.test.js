import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

import { AppRouter } from "../../routers/AppRouter";

const initialState = {
  auth: {
    logged: false,
    userName: "",
  },
};

let store = mockStore(initialState);
store.dispatch = jest.fn();

describe("Pruebas en AppRouter.js", () => {
  test("Debe de mostrar el login si no esta autenticado", () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AppRouter />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
