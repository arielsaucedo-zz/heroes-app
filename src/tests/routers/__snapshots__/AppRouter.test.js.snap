// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter.js Debe de mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="d-flex justify-content-center"
  >
    <form
      className="box"
      onSubmit={[Function]}
    >
      <h1>
        Login
      </h1>
      <p
        className="text-muted"
      >
         Please enter your login and password!
      </p>
      <input
        autoComplete="off"
        name="email"
        onChange={[Function]}
        placeholder="Username"
        type="email"
        value="challenge@alkemy.org"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="react"
      />
      <input
        className="btn-login"
        disabled={false}
        name=""
        type="submit"
        value="Login"
      />
    </form>
  </div>
</div>
`;
